src/metrics/prometheus.js

```diff
@@ -0,0 +1,94 @@
+// @flow
+// Copyright (c) 2018 Jaeger Author.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under
+// the License.
+
+import { Counter as PromCounter, Gauge as PromGauge } from 'prom-client';
+
+let CounterPromWrapper = class {
+  _counter: PromCounter;
+  _labels: any;
+
+  constructor(counter: PromCounter, labels: any) {
+    this._counter = counter;
+    this._labels = labels;
+  }
+
+  increment(delta: number): void {
+    this._counter.inc(this._labels, delta);
```

Comment:
is it possible to push the labels to construction time only? Something like `new Counter(..).withLabels(...)`. Otherwise it seems like a rather expensive call.

Reply:
Could you let me know why it is rather expensive call? 
Label is an object that is pass by reference. Can it be better if the labels arg is removed in constructor?

Reply:
I don't know how exactly prom client implements it, but for example our internal metrics libs that support tagging need to derive a single "metric id" used as a key/reference to the counter being incremented. So we always (as much as possible) apply labels at construction of the Counter object, so that it can have the direct point to the underlying counter that's being incremented, i.e. a call to `counter.inc()` is almost the same as some atomic `c += 1` under the hood. In contrast, when the labels are passed to each call to `inc()` the framework needs to redo the resolution of the metric id and locate that atomic counter to increment - orders of magnitude more expensive op.

Reply:
Pushing lable at constuction time is done.
https://github.com/eundoosong/jaeger-client-node/blob/425d75be6baefd4280eac90e0404db510fe02122/src/metrics/prometheus.js#L63